Core/src/org/sleuthkit/autopsy/communications/VisualizationPanel.java

```diff
@@ -674,7 +715,116 @@ private void fitGraph() {
+        panel.add(new JLabel(Bundle.VisualizationPane_fileName_prompt()));
+        panel.add(text);
+
+        text.setText(defaultReportName);
+
+        int result = JOptionPane.showConfirmDialog(graphComponent, panel,
+                Bundle.VisualizationPanel_action_dialogs_title(), JOptionPane.OK_CANCEL_OPTION);
+
+        List<CommunicationsFilter.SubFilter> filters = currentFilter.getAndFilters();
+
+        if (result == JOptionPane.OK_OPTION) {
+            String enteredReportName = text.getText();
+            String reportName = StringUtils.defaultIfBlank(enteredReportName, defaultReportName);
+            Path reportPath = Paths.get(currentCase.getReportDirectory(), reportName);
+            if (!Files.exists(reportPath)) {
+                createReport(currentCase, reportName);
+            } else {
+                String message = String.format("Overwrite existing report?\n%s", reportName);
+                result = JOptionPane.showConfirmDialog(graphComponent, message,
+                        Bundle.VisualizationPanel_action_dialogs_title(), JOptionPane.OK_CANCEL_OPTION);
 
+                if (result == JOptionPane.OK_OPTION) {
+                    FileUtil.deleteFileDir(reportPath.toFile());
+                    createReport(currentCase, reportName);
+                }
+            }
+        }
+    }
+    
+    /**
+     * Create the Snapshot Report.
+     * 
+     * @param currentCase The current case
+     * @param reportName User selected name for the report
+     * 
+     * @throws IOException 
+     */
+    @NbBundle.Messages({
+        "VisualizationPane_DisplayName=Open Report",
+        "VisualizationPane_NoAssociatedEditorMessage=There is no associated editor for reports of this type or the associated application failed to launch.",
+        "VisualizationPane_MessageBoxTitle=Open Report Failure",
+        "VisualizationPane_NoOpenInEditorSupportMessage=This platform (operating system) does not support opening a file in an editor this way.",
+        "VisualizationPane_MissingReportFileMessage=The report file no longer exists.",
+        "VisualizationPane_ReportFileOpenPermissionDeniedMessage=Permission to open the report file was denied.",
+        "# {0} -  report path",
+        "VisualizationPane_Report_Success=Report Successfully create at:\n{0}",
+        "VisualizationPane_Report_OK_Button=OK",
+        "VisualizationPane_Open_Report=Open Report",})
+    private void createReport(Case currentCase, String reportName) throws IOException {
+        
+        // Create the report.
+        Path reportFolderPath = Paths.get(currentCase.getReportDirectory(), reportName, Bundle.VisualizationPane_reportName()); //NON_NLS
+        BufferedImage image = mxCellRenderer.createBufferedImage(graph, null, graph.getView().getScale(), Color.WHITE, true, null);
+        Path reportPath = (new CommSnapShotReportWriter(currentCase, reportFolderPath, reportName, new Date(), image, currentFilter)).writeReport();
```

Comment:
There is a confusing extra set of parentheses on line 791

Reply:
I think they add extra clarity by grouping the creations of the object from the call. 

Like 2*4+4 is fine, but (2*4)+4 is a little easier read.